src/main/java/org/kohsuke/github/GitHubConnectorResponseErrorHandler.java

```diff
@@ -59,8 +65,38 @@ public void onError(@NotNull GitHubConnectorResponse connectorResponse) throws I
             if (connectorResponse.statusCode() == HTTP_NOT_FOUND) {
                 throw new FileNotFoundException(connectorResponse.request().url().toString());
             } else {
-                throw new HttpException(connectorResponse);
+                HttpException exception = new HttpException(connectorResponse);
+                if (isGitHubDown(exception, connectorResponse.header(CONTENT_TYPE))) {
+                    throw new ServiceDownException(exception);
+                }
+                throw exception;
```

Comment:
Could you talk for a minute why you like this structure instead of the one I suggested? 
Also is there a reason not to check `statusCode() == HTTP_UNAVAILABLE` or at least `statusCode() >= HTTP_INTERNAL_ERROR`?   
```
            if (connectorResponse.statusCode() == HTTP_NOT_FOUND) {
                throw new FileNotFoundException(connectorResponse.request().url().toString());
            } else if (isServiceDown(connectorResponse)) {
                throw new ServiceDownException(connectorResponse);
            } else { 
                throw new HttpException(connectorResponse);
            }
```

I'm not completely opposed to what you have here, but it looks odd to me.

Reply:
I am not really 100% sure, if it's 5xx status code. I do suspect that, but I do not have the evidence, to back me up. That's why I didn't assume it previously. But I will add it there. For the test case I updated the status code manually from 404. I have mentioned this in the original issue as well.
I am not able to invoke at will GitHub's response, when it's down.